crimsobot/cogs/mystery.py

```diff
@@ -4,16 +4,106 @@
 from discord.ext import commands
 
 from crimsobot.bot import CrimsoBOT
+from crimsobot.exceptions import NoMatchingTarotCard
 from crimsobot.utils import tarot
 from crimsobot.utils import tools as c
-from crimsobot.utils.tarot import Deck, Suit
+from crimsobot.utils.tarot import Card, Deck, Suit
 
 
 class Mystery(commands.Cog):
     def __init__(self, bot: CrimsoBOT):
         self.bot = bot
 
+    async def prompt_for_single_card(self, ctx: commands.Context) -> Card:
+        # the suits
+        suits = [s for s in Suit]
+        suit_list = []
+        for idx, suit in enumerate(suits):
+            suit_list.append('{}. {}'.format(idx + 1, suit))
+
+        # prompt 1 of 2: choose suit
+        embed = c.crimbed(
+            title='Choose a suit:',
+            descr='\n'.join(suit_list),
+            thumb_name='wizard',
+            footer='Type the number to choose.'
+        )
+        prompt_suit = await ctx.send(embed=embed)
+
+        # define check for suit
+        def suit_check(msg: discord.Message) -> bool:
+            try:
+                valid_choice = 0 < int(msg.content) <= len(suits)
+                in_channel = msg.channel == ctx.message.channel
+                is_author = msg.author == ctx.message.author
+
+                return valid_choice and in_channel and is_author
+
+            except ValueError:
+                return False
+
+        # wait for user to spcify suit
+        try:
+            msg = await self.bot.wait_for('message', check=suit_check, timeout=45)
+        except asyncio.TimeoutError:
+            await prompt_suit.delete()
+            raise NoMatchingTarotCard
```

Comment:
why not just `raise` so we re-raise and then catch `asyncio.TimeoutError` in `card()`?
same applies for prompt 2.

Reply:
Good point. Will update.